.github/workflows/checks.yml

```diff
@@ -0,0 +1,126 @@
+      adapter-list: ${{ steps.changed-adapters.outputs.CHANGED_ADAPTERS }}
+    steps:
+      - name: Check out code
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+      - name: Set up and install dependencies
+        uses: ./.github/actions/setup
+        with:
+          upstream-branch: ${{ env.UPSTREAM_BRANCH }}
+      - name: Build list of changed packages and changed adapters
+        id: changed-adapters
+        run: ./.github/scripts/changed-adapters.sh
+
+  # ---------- Adapter checks (only for changed EAs) ----------
+
+  # Run unit tests
+  unit-tests:
+    name: Run unit tests for changed adapters
+    runs-on: [self-hosted, sdlc-ghr-prod]
+    needs:
+      - check-changesets
+      - install-packages
+    steps:
+      - name: Check out code
+        uses: actions/checkout@v3
+      - name: Set up and install dependencies
+        uses: ./.github/actions/setup
+      - name: Run unit tests
+        env:
+          CHANGED_PACKAGES: ${{ needs.install-packages.outputs.changed-packages }}
+        run: EA_PORT=0 METRICS_ENABLED=false yarn jest $(echo $CHANGED_PACKAGES | jq '.[].location + "/test/unit"' -r | tr '\n' ' ')
+
+  # Run integration tests
+  integration-tests:
+    name: Run integration tests for changed adapters
+    runs-on: [self-hosted, sdlc-ghr-prod]
+    needs:
+      - check-changesets
+      - install-packages
+    steps:
+      - name: Check out code
+        uses: actions/checkout@v3
+      - name: Set up and install dependencies
+        uses: ./.github/actions/setup
+      - name: Run integration tests
+        env:
+          CHANGED_PACKAGES: ${{ needs.install-packages.outputs.changed-packages }}
+        run: EA_PORT=0 METRICS_ENABLED=false yarn jest $(echo $CHANGED_PACKAGES | jq '.[].location + "/test/integration"' -r | tr '\n' ' ')
+
+  # Run linters
+  linters:
+    name: Run linters and formatters
+    runs-on: [self-hosted, sdlc-ghr-prod]
+    needs:
+      - check-changesets
+      - install-packages
+    steps:
+      - name: Check out code
+        uses: actions/checkout@v3
+      - name: Set up and install dependencies
+        uses: ./.github/actions/setup
+      - name: Lint all files
+        run: yarn lint
+      - name: Check for formatting errors
+        run: yarn format:check
+
+  # Run documentation tests (check that the doc generation succeeds to avoid problems downstream)
+  run-documentation-check:
+    name: Documentation generation test
+    runs-on: [self-hosted, sdlc-ghr-prod]
+    needs:
+      - check-changesets
+      - install-packages
+    env:
+      METRICS_ENABLED: false
+    steps:
+      - name: Check out code
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+      - name: Set up and install dependencies
+        uses: ./.github/actions/setup
+      - name: Test Master List Generation
+        run: yarn generate:master-list -v
+      - name: Test README Generation
+        run: yarn generate:readme -v
```

Comment:
You may be able to use `CHANGED_PACKAGES` here as well

Reply:
The scripts already use the upstream branch to make the calculations, it's a bit inefficient since we could use the packages like you say but changing the code was too much for this PR imo